Use of undefined member m_engine_size in library/src/rng/xorwow.hpp

**Describe the bug**
xorwow_generator_template::operator=(xorwow_generator_template&& other) has the line:
        m_engines_size                 = other.m_engine_size;
but m_engine_size appears nowhere else in the source. It should probably be "other.m_engines_size".

A recent clang change (https://github.com/llvm/llvm-project/pull/90152) fixes a bug which caused this code to previously be erroneously accepted.

**To Reproduce**
Build rocRAND with clang that includes the above fix.

**Expected behavior**
Build is successful.

**Log-files**
In file included from /work/anjenner/reland/rocRAND/library/src/rng/generator_type_xorwow.cpp:23:
/work/anjenner/reland/rocRAND/library/src/rng/xorwow.hpp:204:48: error: no member named 'm_engine_size' in 'xorwow_generator_template<System, ConfigProvider>'; did you mean 'm_engines_size'?
  204 |         m_engines_size                 = other.m_engine_size;
      |                                                ^~~~~~~~~~~~~
      |                                                m_engines_size
/work/anjenner/reland/rocRAND/library/src/rng/xorwow.hpp:426:18: note: 'm_engines_size' declared here
  426 |     unsigned int m_engines_size        = 0;
      |                  ^
1 error generated when compiling for gfx1030.
make[2]: *** [library/CMakeFiles/rocrand.dir/build.make:272: library/CMakeFiles/rocrand.dir/src/rng/generator_type_xorwow.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:880: library/CMakeFiles/rocrand.dir/all] Error 2
make: *** [Makefile:156: all] Error 2

--- a/library/src/rng/xorwow.hpp
+++ b/library/src/rng/xorwow.hpp
@@ -201,7 +201,7 @@ public:
         m_engines_initialized          = other.m_engines_initialized;
         m_engines                      = other.m_engines;
         m_start_engine_id              = other.m_start_engine_id;
-        m_engines_size                 = other.m_engine_size;
+        m_engines_size                 = other.m_engines_size;
         m_seed                         = other.m_seed;
         m_poisson                      = std::move(other.m_poisson);
 

